### PR TITLE
Layermetadata

### PR DESCRIPTION
**Pull request by KaiVolland**

### CODE REVIEW

File: app/model/Layer.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ Ext.define('MoMo.admin.model.Layer', {
         name: 'name',
         type: 'string'
     }, {
-        name: 'type',
+        name: 'hoverFieldTemplate',
```
Comment (weskamm) [inline app/model/Layer.js]: are you sure with this change? seems to be unrelated

File: app/util/Metadata.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+        /**
+         *
+         */
+        getInsertBlankXml: function(){
+            var xml;
+            Ext.Ajax.request({
+                url: BasiGX.util.Url.getWebProjectBaseUrl() + '/admin/resources/data/xmlTemplates/insertMetadata.xml',
+                method: 'GET',
+                async: false,
```
Comment (weskamm) [inline app/util/Metadata.js:87]: why?

File: app/util/Metadata.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+            return xml;
+        },
+
+        /**
+         *
+         */
+        uuidFromXmlString: function(xmlString){
+            var xml = this.parseXml(xmlString);
+            var identifierNode = xml.getElementsByTagName('identifier')[0];
+            return identifierNode.innerHTML;
```
Comment (weskamm) [inline app/util/Metadata.js]: could fail beacuse of undefined identifierNode

File: app/util/Metadata.js [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        getTextContentsViaQuerySelector: function(xml, selectorString) {
+            if(!xml){
+                throw "No xml document given.";
+            }
+            if(!selectorString){
+                throw "No selectorString given.";
+            }
+
+            var elements = xml.documentElement.querySelectorAll(selectorString);
+            return Ext.Array.pluck(elements, "textContent");
```
Comment (weskamm) [inline app/util/Metadata.js]: could fail beacuse of undefined elements

File: app/view/panel/layer/MetadataController.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                var uuid;
+                if (response && response.responseText) {
+                    responseObj = Ext.decode(response.responseText);
+                    uuid = MoMo.admin.util.Metadata.uuidFromXmlString(
+                            responseObj.data);
+                    var layer = me.getView().lookupViewModel().get('layer');
+                    layer.set('metadataIdentifier', uuid);
+                    layer.save();
+                }
+                Ext.toast('Createad MetadataSet with UUID: ' + uuid);
```
Comment (weskamm) [inline app/view/panel/layer/MetadataController.js:36]: errorhandling: maybe use callback instead of success and handle error / fail or move the toast into if and add one for else case

File: app/view/panel/layer/MetadataController.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                'metadata/csw.action',
+                method: "POST",
+                params: {
+                    xml: xml
+                },
+                defaultHeaders: BasiGX.util.CSRF.getHeader(),
+                scope: this,
+                success: function() {
+                    Ext.toast('Updated MetadataSet with UUID: ' + uuid);
+                }
```
Comment (weskamm) [inline app/view/panel/layer/MetadataController.js:71]: failure?

File: app/view/tab/CreateOrEditLayerController.js [hunk 7]
```diff
@@ -48,12 +56,35 @@ Ext.define('MoMo.admin.view.tab.CreateOrEditLayerController', {
...
+            params: {
+                xml: MoMo.admin.util.Metadata.getLoadXml(uuid)
+            },
+            defaultHeaders: BasiGX.util.CSRF.getHeader(),
+            success: function(response){
+                var responseObj = Ext.decode(response.responseText);
+                var metadataObj = MoMo.admin.util.Metadata.parseMetadataXml(
+                        responseObj.data);
+                viewModel.set('metadata', metadataObj);
+            }
```
Comment (weskamm) [inline app/view/tab/CreateOrEditLayerController.js:80]: failure?

**Comment by weskamm** [comment]
added some comments, mostly error handling. Nice PR!

File: app/model/Layer.js (hunk 1 above)
Comment (KaiVolland) [reply app/model/Layer.js]: I'm not 100% sure about this. I think that "type" is not exisiting in the backend so i removed it. Hovertemplate seems to be in the Appearance ... so i'll remov this.

File: app/util/Metadata.js (hunk 2 above)
Comment (KaiVolland) [inline app/util/Metadata.js]: It returns the UUID which is needed as a property for the layer to create. I tried to avoid nesting code, but this might break the layer creation. I'll fix this.

File: app/model/Layer.js (hunk 1 above)
Comment (weskamm) [reply app/model/Layer.js]: yes template has moved to appearance. and the layer should have a field `dataType`.
Maybe you can check if upload for raster and vector is still working after merge, becuase thats related to that

File: app/util/Metadata.js (hunk 2 above)
Comment (weskamm) [reply app/util/Metadata.js:87]: (async)

Comment (KaiVolland) [reply app/util/Metadata.js:87]: This is a request to a static file which is needed for the real request. Maybe there is a better/simpler way to do this... we should keep this in mind.

Comment (weskamm) [reply app/util/Metadata.js:87]: no i just meant why did you request it synchronous instead of async?

Comment (KaiVolland) [reply app/util/Metadata.js:87]: I get that. I just didn't know the simplest way to just get the template xml from the resources... I could do this asynchronous but this keeps it more capsulated.

File: app/model/Layer.js (hunk 1 above)
Comment (KaiVolland) [reply app/model/Layer.js]: Tested with raster and vetor data.
